Data/ByteString.hs

```diff
@@ -1078,6 +1080,20 @@ index ps n
     | otherwise      = ps `unsafeIndex` n
 {-# INLINE index #-}
 
+-- | 'ByteString' index that returns 'Nothing' if:
```

Comment:
An `/O(1)/` is missing to match `index`'s haddocks; also it should mention that it's 0-based (like `index`'s docstring does).